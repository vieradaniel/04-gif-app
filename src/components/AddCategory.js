import React, { useState } from 'react';

export const AddCategory = () => {

    const [InputValue, setInputValue] = useState("Hola Mundo");
    const handleInputChange = (e)=>{
        console.log(e.target.value);
        setInputValue(e.target.value);
    }
    const handleSubmit = (e)=>{
        e.preventDefault();
        console.log('Submit hecho');
    }

  return (
        
        <form onSubmit={handleSubmit}>
            <input 
                type="text" 
                value = { InputValue }
                onChange = { handleInputChange }
            />
        </form>
        
        );
};
