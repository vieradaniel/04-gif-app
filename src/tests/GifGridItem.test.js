import React from 'react';

import'@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';



describe('pruebas en <GifGridItem/>',()=>{

    const url="http://localhost/algo.jpg";
    const title="un titulo";
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    
    test('debe de mostrar <GifGridItem/> correctamente',()=>{


        expect(wrapper).toMatchSnapshot();


    })

    test('debe de tener un pÃ¡rrafo con el title',()=>{

        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    })

    test('debe tener la imagen y la url de los props',()=>{

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url) ;
        expect(img.prop('alt')).toBe(title) ;

    });

    test('debe de tener animate__fadeIn',()=>{

        const div = wrapper.find('div');
        
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);

    })



})