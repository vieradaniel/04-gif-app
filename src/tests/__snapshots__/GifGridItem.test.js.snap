// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem/> debe de mostrar <GifGridItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
